go/vm/lfvm/interpreter_test.go

```diff
@@ -69,24 +69,39 @@ func getContext(code Code, data []byte, runContext vm.RunContext, stackPtr int,
 
 // Test UseGas function and correct status after running out of gas
 func TestGasFunc(t *testing.T) {
-	ctx := getEmptyContext()
-	ctx.gas = 100
-	ok := ctx.UseGas(10)
-	if !ok {
-		t.Errorf("expected not failed useGas function, got failed")
-	}
-	if ctx.gas != 90 {
-		t.Errorf("expected gas in context is 90, got %d", ctx.gas)
-	}
-	ok = ctx.UseGas(100)
-	if ok {
-		t.Errorf("expected failed useGas function, got ok")
-	}
-	if ctx.gas != 90 {
-		t.Errorf("expected gas in context is 90 also after failing, got %d", ctx.gas)
+
+	tests := []struct {
+		name      string
```

Comment:
nit: when a case has a name it is custom to use the name as the key of a map; something like this:
```

	tests := map[string]struct {
		gas       vm.Gas
		amount    vm.Gas
		wantedGas vm.Gas
		expected  bool
		status    Status
	}{
		"Zero amount":       {100, 0, 100, true, RUNNING},
		"Sufficient gas":    {100, 10, 90, true, RUNNING},
		"Insufficient gas":  {10, 100, 10, false, OUT_OF_GAS},
		"All gas":           {100, 100, 0, true, RUNNING},
		"Negative amount":   {100, -100, 200, true, RUNNING},
		"Negative gas":      {-100, 100, -100, false, OUT_OF_GAS},
		"Negative Negative": {-100, -100, -100, false, OUT_OF_GAS},
	}

	for name, test := range tests {
            ...
        }
```